src/main/scala/org/broadinstitute/dsde/firecloud/model/ModelJsonProtocol.scala

```diff
@@ -104,6 +104,59 @@ object ModelJsonProtocol {
     }
   }
 
+  implicit object impLibrarySearchParams extends RootJsonFormat[LibrarySearchParams] {
+    val SEARCH_TERM = "searchTerm"
+    val FROM = "from"
+    val SIZE = "size"
+
+    override def write(params: LibrarySearchParams): JsValue = params.searchTerm match {
+      case None => JsObject(Map(FROM -> JsNumber(params.from), SIZE -> JsNumber(params.size)))
+      case Some(term) => JsObject(Map(SEARCH_TERM -> JsString(term), FROM -> JsNumber(params.from), SIZE -> JsNumber(params.size)))
+    }
+
+    override def read(json: JsValue): LibrarySearchParams = {
+      val data = json.asJsObject.fields
+      val term = data.getOrElse(SEARCH_TERM, None) match {
+        case JsString(str) => Some(str)
+        case None => None
+        case _ => throw DeserializationException("unexpected json type for " + SEARCH_TERM)
+      }
+      val from: Option[Int] = data.getOrElse(FROM, None) match {
+        case JsNumber(f) => Some(f.intValue)
+        case None => None
+        case _ => throw DeserializationException("unexpected json type for " + FROM)
+      }
+      val size: Option[Int] = data.getOrElse(SIZE, None) match {
+        case JsNumber(s) => Some(s.intValue)
+        case None => None
+        case _ => throw DeserializationException("unexpected json type for " + SIZE)
+      }
+
+      (from, size) match {
+        case (None, None) => LibrarySearchParams(term)
+        case (Some(f), None) => LibrarySearchParams(term, f)
+        case (None, Some(s)) => LibrarySearchParams(term, size=s)
+        case (Some(f), Some(s)) => LibrarySearchParams(term, f, s)
+      }
```

Comment:
could you replace the entire `(from, size) match` with `LibrarySearchParams(term, from, size)`, and then inside of ModelJsonProtocol, have:
```
case class LibrarySearchParams(
  searchTerm: Option[String],
  from: Int = 0,
  size: Int = 10)

object LibrarySearchParams {
  def apply(searchTerm: Option[String], from: Option[Int], size: Option[Int]) = {
    new LibrarySearchParams(searchTerm, from.getOrElse(0), size.getOrElse(10))
  }
}
```

this does mean you have the defaults (0,10) in two places, but they're within a few lines of each other in the class and its companion. Disclaimer: I haven't actually tried this.